frontend/store/index.js

```diff
@@ -0,0 +1,188 @@
+      if (!context.state.hasRegisteredEvents) {
+        window.ethereum.on('chainChanged', (chainId) => {
+          if (Number(chainId) !== Number(this.$config.chain_id)) {
+            this.commit('setIsCorrectChain', false)
+          } else {
+            this.commit('setIsCorrectChain', true)
+          }
+        })
+        this.commit('setRegisteredEvents', true)
+      }
+      return true
+    } else {
+      this.commit('setWeb3', null)
+      return false
+    }
+  },
+  // Request account access from user
+  async safelyRequestAccounts (context) {
+    if (await !this.dispatch('checkHasProvider')) {
+      return false
+    }
+    return await this.dispatch('_requestAccounts')
+  },
+  async _switchChains (context) {
+    const ethereum = window.ethereum
+    try {
+      await ethereum.request({
+        method: 'wallet_switchEthereumChain',
+        params: [{ chainId: this.$config.chain_id }]
+      })
+    } catch (switchError) {
+      // This error code indicates that the chain has not been added to MetaMask.
+      if (switchError.code === 4902) {
+        await ethereum.request({
+          method: 'wallet_addEthereumChain',
+          params: [
+            {
+              chainId: this.$config.chain_id,
+              chainName: 'Eleos Chain',
+              rpcUrls: [this.$config.ganache_url]
+            }
+          ]
+        })
+      } else {
+        throw switchError
+      }
+    }
+    return true
+  },
+  async _requestAccounts (context) {
+  // Request account access
+    try {
+      const accounts = await window.ethereum.request({ method: 'eth_requestAccounts' })
+
+      // Connected if accounts is not empty
+      if (accounts.length > 0) {
+        this.commit('setConnected', true)
+        this.commit('setPreviouslyConnected', true)
+        return accounts
+      } else {
+        Notification.open({
+          type: 'is-warning',
+          message: 'No accounts found within your wallet!'
+        })
+        this.commit('setConnected', false)
+        this.commit('setPreviouslyConnected', false)
+        return false
+      }
+    } catch (error) {
+      console.error(error)
+      Notification.open({
+        type: 'is-danger',
+        message: 'Failed to connect to your wallet!'
+      })
+      this.commit('setConnected', false)
+      this.commit('setPreviouslyConnected', false)
+      return false
+    }
+  },
+  async callToCampaignFactory (
+    context,
+    {
+      campaignName,
+      beneficiaryUrl,
+      endDate,
+      beneficiaryAddress,
+      campaignOwnerAddress,
+      targetAmount
+    }
+  ) {
+    // Format date to timestamp
+    let timestamp
+    if (endDate !== null) {
+      timestamp = Date.parse(endDate) / 1000
+    } else {
+      // Set an "impossible" end date
+      timestamp = String(Math.pow(2, 63) - 1)
+    }
+
+    // 1 eth = 1000000000000000000 wei
+    const targetAmountInWei = targetAmount * 1000000000000000000
```

Comment:
Not sure if this will cause an overflow. Eth values in Wei are pretty huge. Might want to consider using BN.js or in string?

Reply:
On line 180, the targetAmountInWei is already converted to String with the String() function. However, I can move over the String() function to line 159 instead tomorrow morning.

Reply:
I think there might still be some issues if target amount is large because the value before assignment to `targetAmountInWei` could overflow. The largest safe int value in JS is [9007199254740991](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER) and even with 1 eth target amount it already exceeds it.

Reply:
Changed to used web3.utils to convert it safely